src/Control/Monad/Freer/TH.hs

```diff
@@ -0,0 +1,128 @@
+{-# LANGUAGE LambdaCase                        #-}
+{-# LANGUAGE OverloadedStrings                 #-}
+{-# LANGUAGE TemplateHaskell                   #-}
+{-# OPTIONS_GHC -fno-warn-missing-import-lists #-}
+
+-- This has been ported from https://github.com/IxpertaSolutions/freer-effects/pull/28,
+-- where all the work has been done by @isovector (Sandy Maguire).
+
+-- | Automatic generation of freer monadic actions.
+module Control.Monad.Freer.TH
+  ( makeFreer
+  , makeFreer_
+  )
+where
+
+import Control.Monad (forM, unless)
+import Control.Monad.Freer (send, Member, Eff)
+import Data.Char (toLower)
+import Data.List (nub)
+import Data.Maybe (mapMaybe)
+import Language.Haskell.TH
+import Prelude
+
+
+-- | @$('makeFreer' ''T)@ provides freer monadic actions for the constructors of
+-- the given GADT @T@.
+makeFreer :: Name -> Q [Dec]
+makeFreer = genFreer True
```

Comment:
Can you rename this to `makeEffect` instead of `makeFreer`? We’re not really generating a freer monad here, we’re defining a new effect.